src/inet/InetInterface.cpp

```diff
@@ -502,6 +502,142 @@ void CloseIOCTLSocket()
     }
 }
 
+#if __ANDROID__
+
+static struct if_nameindex * backport_if_nameindex(void);
+static void backport_if_freenameindex(struct if_nameindex *);
+
+static void backport_if_freenameindex(struct if_nameindex * inArray)
+{
+    if (inArray == nullptr)
+    {
+        return;
+    }
+
+    for (size_t i = 0; inArray[i].if_index != 0; i++)
+    {
+        if (inArray[i].if_name != nullptr)
+        {
+            Platform::MemoryFree(inArray[i].if_name);
+        }
+    }
+
+    Platform::MemoryFree(inArray);
+}
+
+static struct if_nameindex * backport_if_nameindex(void)
+{
+    int err;
+    unsigned index;
+    size_t intfIter              = 0;
+    size_t maxIntfNum            = 0;
+    size_t numIntf               = 0;
+    size_t numAddrs              = 0;
+    struct if_nameindex * retval = nullptr;
+    struct if_nameindex * tmpval = nullptr;
+    struct ifaddrs * addrList    = nullptr;
+    struct ifaddrs * addrIter    = nullptr;
+    const char * lastIntfName    = "";
+
+    err = getifaddrs(&addrList);
+    VerifyOrExit(err >= 0, );
+
+    // coalesce on consecutive interface names
+    for (addrIter = addrList; addrIter != nullptr; addrIter = addrIter->ifa_next)
+    {
+        numAddrs++;
+        if (strcmp(addrIter->ifa_name, lastIntfName) == 0)
+        {
+            continue;
```

Comment:
I think we you means

        if (strcmp(addrIter->ifa_name, lastIntfName) )
        {
            numIntf++;
            lastIntfName = addrIter->ifa_name;
        }

Reply:
I modified.